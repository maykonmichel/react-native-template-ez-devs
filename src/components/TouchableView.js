import React from 'react';
import { Platform, TouchableNativeFeedback, TouchableOpacity, View } from 'react-native';

const isAndroid = Platform.OS === 'android';

const Touchable = isAndroid ? TouchableNativeFeedback : TouchableOpacity;

const TouchableView = ({ children, style, ...attributes }) => (
  <Touchable style={!isAndroid && style} {...attributes}>
    <View style={isAndroid && style}>{children}</View>
  </Touchable>
);

TouchableView.propTypes = Touchable.propTypes;

TouchableView.defaultProps = Touchable.getDefaultProps();

export default React.memo(TouchableView);
