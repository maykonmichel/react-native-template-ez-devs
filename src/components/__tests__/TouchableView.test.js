import React from 'react';
import { shallow } from 'enzyme';

import TouchableView from '../TouchableView';

describe('Testing Text component', () => {
  const touchableView = shallow(<TouchableView>Test</TouchableView>);
  it('renders as expected', () => {
    expect(touchableView).toMatchSnapshot();
  });
  it('calls function on press', () => {
    const onPress = jest.fn();
    touchableView.setProps({ onPress });
    touchableView.simulate('press');
    expect(onPress.mock.calls.length).toBe(1);
  });
});
