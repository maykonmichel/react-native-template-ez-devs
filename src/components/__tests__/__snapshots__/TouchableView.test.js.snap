// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TouchableView component renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <View
    style={false}
  >
    Test
  </View>
</TouchableOpacity>
`;
